### PR TITLE
PromCon 2017

### PR DESCRIPTION
**Pull request by grobie**
Done:

* Update all dependencies
* Upgrade to nanoc4
* Clean up layouts (reduce duplication of layout code)
* Prepare some code paths for multiple PromCon conferences
* Add first stub of PromCon 2017

TODO:

* [ ] Re-implement cache buster
* [x] Replace dummy munich banner image (can't really browse flickr with the slow connection here)
* [x] Update register page (form URL and admission price range)
* [ ] Update sponsor page
  * [x] form URL
  * [ ] sponsorship table
  * [ ] ecosystem numbers
  * [ ] ToS
* [ ] Update CfP page
  * [x] form URL
  * [ ] dates
  * [ ] topics
* [x] Link back to 2016 page (from the footer?)

@juliusv @brian-brazil

### CODE REVIEW

File: content/2017-munich/register.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: Register
+---
+
+## Register to attend
+
+Attendee registration will open soon — fill out the form linked below to get
+notified when the registration opens:
+
+<a class="btn btn-lg btn-default" href="https://docs.google.com/forms/d/12H6o3vvnnQUJSDV1mQ77EqpU9B64Jk70iFboHwb2y3M/viewform" target="_blank" role="button"><i class="fa fa-briefcase"></i>&nbsp;&nbsp;Register</a>
```
Comment (juliusv) [inline content/2017-munich/register.md]: That's still the old form (it says 2016) - did you intend to link a new one already?

Comment (grobie) [inline content/2017-munich/register.md]: No. See the todos.

Comment (juliusv) [inline content/2017-munich/register.md]: Ah ok, it's still listed as a TODO, sorry.

File: layouts/kind/talk.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<h3>Talk abstract</h3>
+<%= yield %>
+<a class="btn btn-default" href="/2016-berlin/schedule/">Back to schedule</a>
```
Comment (juliusv) [inline layouts/kind/talk.html]: Should there be a TODO for enabling 2017 versions of the speaker and talk pages?

File: layouts/container.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="col-md-8">
+      <div class="content">
+        <%= yield %>
+      </div>
+
+      <hr>
+
+      <footer>
+        <p class="pull-left">
+          &copy; Prometheus Authors 2016
```
Comment (juliusv) [inline layouts/container.html]: Just update the copyright generally to 2017 I'd say.

**Comment by juliusv** [comment]
- There should be *some* link somewhere to the old 2016 conference site (for looking at talks etc.), otherwise you can only get to it if you already have an old link.

- Is it intentional that speaker and talk pages now also have the banner on top? (the original intention for them having no banner is to not annoy people who just want to quickly look at a talk's details on their phone)

**Comment by juliusv** [comment]
Great otherwise :)

File: layouts/container.html (hunk 3 above)
Comment (grobie) [inline layouts/container.html]: Ok.

Side note: for source code files, I recently read the year should describe the creation of a file, not the last modification.

File: layouts/kind/talk.html (hunk 2 above)
Comment (grobie) [inline layouts/kind/talk.html]: I'd go live with the 2017 page without speaker and talks, given we won't have any in the beginning? It'll probably be a  new task to create these pages once the CfP is over..

**Comment by grobie** [comment]
I added a TODO for the link back to 2016. The change was kind of intentional, it looked so boring on my laptop screen. It's an issue on mobile though, I agree.

**Comment by grobie** [comment]
I changed the behavior back to the original behavior (no more banner on talk/speaker pages). This would be better solved with device-dependent css though.

**Comment by juliusv** [comment]
All sounds agreeable to me :)

**Comment by juliusv** [comment]
@grobie just since is needed for CfP and sponsorship acquisition, what's your rough time plan for this? I'm happy to finish updating the sponsors and CfP page contents for 2017 if you just leave them disabled from the nav for now.

**Comment by grobie** [comment]
I have a new cache buster ready and can publish today. I was waiting for
all the missing content.

**Comment by juliusv** [comment]
Awesome. Yeah, I can update the content pages (CfP + sponsors) later this week.

**Comment by grobie** [comment]
Alright, I merged the current state and removed CfP and sponsors pages from the menu. Feel free to update and uncomment in `lib/helpers/navigation.rb`.

**Comment by grobie** [comment]
@juliusv For the ecosystem numbers you could take a look at Brian's keynote slides. I saw some numbers in there.

**Pull request merged**